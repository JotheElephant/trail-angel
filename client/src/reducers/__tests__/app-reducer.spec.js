import appReducer from '../app-reducer';
import geolocationData from '../../../__tests__/fixtures/geolocation-data';
import actionTypes from '../../actions/action-types';

const initialState = {
  isLoaded: false,
  geolocation: null,
  gettingLocation: false
};

test('initializes app', () => {
  expect(appReducer(initialState, { type: actionTypes.INITIALIZE_APP })).toMatchSnapshot();
});

test('fetches geolocation', () => {
  expect(appReducer(initialState, { type: actionTypes.GET_GEOLOCATION })).toMatchSnapshot();
});

test('receives geolocation', () => {
  expect(appReducer(initialState, { type: actionTypes.RECEIVE_GEOLOCATION,
                                    geolocation: geolocationData
                                  }
        )).toMatchSnapshot();
});