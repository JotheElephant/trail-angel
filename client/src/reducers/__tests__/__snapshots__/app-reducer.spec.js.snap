exports[`appReducer fetches geolocation 1`] = `
Object {
  "geolocation": null,
  "gettingLocation": true,
  "isLoaded": false,
}
`;

exports[`appReducer initializes app 1`] = `
Object {
  "geolocation": null,
  "gettingLocation": false,
  "isLoaded": true,
}
`;

exports[`appReducer receives geolocation 1`] = `
Object {
  "geolocation": Object {
    "coords": Object {
      "accuracy": 5,
      "altitude": 0,
      "altitudeAccuracy": -1,
      "heading": -1,
      "latitude": 45.3296,
      "longitude": -121.9112,
      "speed": -1,
    },
    "timestamp": 1486169476048.98,
  },
  "gettingLocation": false,
  "isLoaded": false,
}
`;
